pkg/identity/keystone/keystone.go

```diff
@@ -149,6 +150,41 @@ func (k *KeystoneAuth) processNextItem() bool {
 	return true
 }
 
+func (k *KeystoneAuth) updatePolicies(cm *apiv1.ConfigMap, key string) {
+	glog.Info("ConfigMap created or updated, will update the authorization policy.")
+
+	var policy policyList
+	if err := json.Unmarshal([]byte(cm.Data["policies"]), &policy); err != nil {
+		runtimeutil.HandleError(fmt.Errorf("failed to parse policies defined in the configmap %s: %v", key, err))
+	}
+	if len(policy) > 0 {
+		if _, err := json.MarshalIndent(policy, "", "  "); err != nil {
+			runtimeutil.HandleError(fmt.Errorf("failed to parse policies defined in the configmap %s: %v", key, err))
+		}
+	}
+
+	k.authz.mu.Lock()
+	k.authz.pl = policy
+	k.authz.mu.Unlock()
+
+	glog.Infof("Authorization policy updated.")
+}
+
+func (k *KeystoneAuth) updateSyncConfig(cm *apiv1.ConfigMap, key string) {
+	glog.Info("ConfigMap created or updated, will update the sync configuration.")
+
+	var sc *syncConfig
+	if err := json.Unmarshal([]byte(cm.Data["syncConfig"]), sc); err != nil {
+		runtimeutil.HandleError(fmt.Errorf("failed to parse sync config defined in the configmap %s: %v", key, err))
+	}
+
+	k.authz.mu.Lock()
+	k.authn.syncConfig = sc
+	k.authz.mu.Unlock()
```

Comment:
Not sure if it's the best practice to use mux from a struct when updating fields of another struct.

Reply:
Lol :) You're right! for sure we should lock authn, not authz